zerver/tests/test_event_system.py

```diff
@@ -1244,7 +1244,7 @@ def test_queries(self) -> None:
 
         self.login_user(user)
 
-        with self.assert_database_query_count(40):
+        with self.assert_database_query_count(41):
```

Comment:
I think we get to avoid the extra query if we implement my proposal to just return a single object with both categories.